homeassistant/components/websocket_api/commands.py

```diff
@@ -260,6 +261,42 @@ def handle_get_states(
     connection.send_message(response2)
 
 
+@callback
+@decorators.websocket_command(
+    {
+        vol.Required("type"): "subscribe_entites",
+    }
+)
+def handle_subscribe_entites(
+    hass: HomeAssistant, connection: ActiveConnection, msg: dict[str, Any]
+) -> None:
+    """Handle subscribe entities command."""
+    # Circular dep
+    # pylint: disable=import-outside-toplevel
+    from .permissions import SUBSCRIBE_ALLOWLIST
+
+    if "state_changed" not in SUBSCRIBE_ALLOWLIST and not connection.user.is_admin:
+        raise Unauthorized
+
+    @callback
+    def forward_entity_changes(event: Event) -> None:
+        """Forward entity state changed events to websocket."""
+        if not connection.user.permissions.check_entity(
+            event.data["entity_id"], POLICY_READ
+        ):
+            return
+
+        connection.send_message(messages.cached_state_diff_message(msg["id"], event))
+
+    # We must never await between sending the states and listening for
+    # state changed events or we will introduce a race condition
+    # where some states are missed
+    handle_get_states(hass, connection, msg)
```

Comment:
calling `handle_get_states` will send data in a "result" response when setting up a subscription. That's not something that we don't handle currently. I guess we can add it.